server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala

```diff
@@ -0,0 +1,112 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.mail
+
+import org.apache.james.core.Username
+import org.apache.james.jmap.model.UnsignedInt
+import org.apache.james.mailbox.Role
+import org.apache.james.mailbox.model.MailboxId
+
+final case class MailboxName(name: String) {
+  require(!name.isEmpty, "'name' is mandatory")
+}
+
+case class MayReadItems(value: Boolean) extends AnyVal
+case class MayAddItems(value: Boolean) extends AnyVal
+case class MayRemoveItems(value: Boolean) extends AnyVal
+case class MaySetSeen(value: Boolean) extends AnyVal
+case class MaySetKeywords(value: Boolean) extends AnyVal
+case class MayCreateChild(value: Boolean) extends AnyVal
+case class MayRename(value: Boolean) extends AnyVal
+case class MayDelete(value: Boolean) extends AnyVal
+case class MaySubmit(value: Boolean) extends AnyVal
+
+case class MailboxRights(mayReadItems: MayReadItems,
+                               mayAddItems: MayAddItems,
+                               mayRemoveItems: MayRemoveItems,
+                               maySetSeen: MaySetSeen,
+                               maySetKeywords: MaySetKeywords,
+                               mayCreateChild: MayCreateChild,
+                               mayRename: MayRename,
+                               mayDelete: MayDelete,
+                               maySubmit: MaySubmit)
+
+object MailboxNamespace {
+  def delegated(owner: Username) = DelegatedNamespace(owner)
+
+  def personal = PersonalNamespace
+}
+
+sealed trait MailboxNamespace {
+  def owner: Option[Username]
```

Comment:
what's the purpose of this `owner`?

Reply:
I don't even see any field with name `...namespace` in the spec. Do we really need it?

Reply:
James shows mailboxes shared with a user directly in his account.

Namespace allows to discriminate if the mailbox belongs to the curreznt user or another one.

Reply:
That's why I think we need to do specification work on these James current extensions ^^

Reply:
but do we ever send `owner` field for `PersonalNamespace`?
If not, you don't need to define it in a parent type at all.